core/src/main/java/com/linecorp/armeria/common/DefaultFlagsProvider.java

```diff
@@ -166,9 +166,13 @@ public Integer maxNumConnections() {
     }
 
     @Override
-    public Integer numCommonWorkers() {
-        final int defaultNumCpuCores = Runtime.getRuntime().availableProcessors();
-        return defaultNumCpuCores * 2;
+    public Integer numCommonWorkers(TransportType transportType) {
```

Comment:
Optional) There is a similar case where a flag depends on another flag to resolve the final value like this. https://github.com/line/armeria/pull/5029#discussion_r1278808201
What do you think of a getter method for the composed `FlagsProvider` and set it to `SystemPropertyFlagsProvider`?

For example:
```java
interface FlagsProvider {
    // Needs a better name.
    FlagsProvider composedProvider();
}

...
SystemPropertyFlagsProvider systemProvider = new SystemPropertyFlagsProvider();
final List<FlagsProvider> flagsProviders = ServiceLoader.load(...);
flagsProviders.add(0, systemProvider);
flagsProviders.add(DefaultFlagsProvider.INSTANCE);
systemProvider.setComposedProvider(flagsProviders);

public Integer numCommonWorkers() {
    if (composedProvider().transportType() == TransportType.IO_URING) {
        ...
    }
}
```
One concern is the circular reference between flags. If it looks overkill or dangerous to use, feel free to ignore this comment.

Reply:
It sounds a little bit dangerous to me because it's fairly easy to trigger circular dependency as you said. We could revisit this when we end up with the situation that really needs this.